src/main/kotlin/net/atos/zac/app/informatieobjecten/InformatieObjectenRESTService.kt

```diff
@@ -0,0 +1,675 @@
+import net.atos.zac.flowable.FlowableTaskService
+import net.atos.zac.flowable.TaakVariabelenService
+import net.atos.zac.policy.PolicyService
+import net.atos.zac.policy.PolicyService.assertPolicy
+import net.atos.zac.util.UriUtil
+import net.atos.zac.webdav.WebdavHelper
+import net.atos.zac.websocket.event.ScreenEventType
+import nl.lifely.zac.util.NoArgConstructor
+import org.apache.commons.lang3.BooleanUtils
+import org.apache.commons.lang3.StringUtils
+import org.jboss.resteasy.annotations.providers.multipart.MultipartForm
+import java.io.IOException
+import java.net.URI
+import java.util.UUID
+
+@Singleton
+@Path("informatieobjecten")
+@Consumes(MediaType.APPLICATION_JSON)
+@Produces(MediaType.APPLICATION_JSON)
+@NoArgConstructor
+@Suppress("TooManyFunctions", "LargeClass", "LongParameterList", "TooGenericExceptionThrown")
+class InformatieObjectenRESTService @Inject constructor(
+    private val drcClientService: DrcClientService,
+    private val ztcClientService: ZTCClientService,
+    private val zrcClientService: ZRCClientService,
+    private val zgwApiService: ZGWApiService,
+    private val flowableTaskService: FlowableTaskService,
+    private val taakVariabelenService: TaakVariabelenService,
+    private val ontkoppeldeDocumentenService: OntkoppeldeDocumentenService,
+    private val inboxDocumentenService: InboxDocumentenService,
+    private val enkelvoudigInformatieObjectLockService: EnkelvoudigInformatieObjectLockService,
+    private val eventingService: EventingService,
+    private val zaakInformatieobjectConverter: RESTZaakInformatieobjectConverter,
+    private val informatieobjectConverter: RESTInformatieobjectConverter,
+    private val informatieObjecttypeConverter: RESTInformatieobjecttypeConverter,
+    private val historieRegelConverter: RESTHistorieRegelConverter,
+    private val gerelateerdeZaakConverter: RESTGerelateerdeZaakConverter,
+    private val documentCreatieService: DocumentCreatieService,
+    private val loggedInUserInstance: Instance<LoggedInUser>,
+    private val webdavHelper: WebdavHelper,
+    private val policyService: PolicyService,
+    private val enkelvoudigInformatieObjectDownloadService: EnkelvoudigInformatieObjectDownloadService,
+    private val enkelvoudigInformatieObjectUpdateService: EnkelvoudigInformatieObjectUpdateService,
+    private val officeConverterClientService: OfficeConverterClientService
+) {
+    companion object {
+        private const val MEDIA_TYPE_PDF = "application/pdf"
+        private const val TOELICHTING_PDF = "Geconverteerd naar PDF"
+    }
+
+    @GET
+    @Path("informatieobject/{uuid}")
+    fun readEnkelvoudigInformatieobject(
+        @PathParam("uuid") uuid: UUID,
+        @QueryParam("zaak") zaakUUID: UUID?
+    ): RESTEnkelvoudigInformatieobject =
+        uuid
+            .let(drcClientService::readEnkelvoudigInformatieobject)
+            .let { enkelvoudigInformatieObject ->
+                zaakUUID?.let(zrcClientService::readZaak)?.let {
```

Comment:
```suggestion
                zaakUUID?.let { zrcClientService::readZaak).let {
```

readZaak can never return null